### PR TITLE
[MI-2396] Create request modal and add slash command to open request modal

### PR DESCRIPTION
**Pull request by ayusht2810**
image
image

### CODE REVIEW

File: webapp/src/containers/createRequest/index.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    const mapRequestsToSuggestions = (requests: RequestData[]): Array<Record<string, string>> => requests.map((r) => ({
+        id: r.sys_id,
+        name: r.name,
+        short_description: r.short_description,
+        price: r.price,
+        title: r.category.title,
+        category_id: r.category.sys_id,
+    }));
+
+    // Set the suggestions when the input value of the auto-suggest changes;
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/createRequest/index.tsx:98]:
```suggestion
    // Set the suggestions when the input value of the auto-suggest changes.
```

File: webapp/src/containers/createRequest/index.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    useEffect(() => {
+        const requestsToSuggest = options?.filter((r) => r.name.toLowerCase().includes(autoSuggestValue.toLowerCase())) || [];
+        setSuggestions(mapRequestsToSuggestions(requestsToSuggest));
+    }, [autoSuggestValue]);
+
+    useEffect(() => {
+        // When the request value is reset, reset the caller auto-suggest input as well;
```
Comment (Kshitij-Katiyar) [inline webapp/src/containers/createRequest/index.tsx:105]: Is there need any of semi colons in the end of comments ?
```suggestion
        // Reset the caller auto-suggest input when the request value is reset.
```

Comment (ayusht2810) [reply webapp/src/containers/createRequest/index.tsx]: updated

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/command.go [hunk 3]
```diff
@@ -268,6 +270,16 @@ func (p *Plugin) handleCreateIncident(_ *plugin.Context, args *model.CommandArgs
...
 
+func (p *Plugin) handleCreateRequest(_ *plugin.Context, args *model.CommandArgs, params []string, client Client, _ bool) string {
+	p.API.PublishWebSocketEvent(
+		constants.WSEventOpenCreateRequestModal,
+		nil,
+		&model.WebsocketBroadcast{UserId: args.UserId},
+	)
+
+	return ""
+}
```
Comment (avas27JTG) [inline server/plugin/command.go:281]: why return an empty string if that is of no use?

File: webapp/src/containers/createRequest/index.tsx [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                                    >
+                                        <span className='search-panel__description-header margin-right-10 text-ellipsis'>{header.label}</span>
+                                        <span className='search-panel__description-text channel-text wt-500 text-ellipsis'>{showModalLoader ? <SkeletonLoader/> : request[header.key] || 'N/A'}</span>
+                                    </li>
+                                ))}
+                            </ul>
+                            {serviceNowBaseURL && (
+                                <div>
+                                    <a
+                                        className='color--link btn btn-primary servicenow-request-button'
```
Comment (avas27JTG) [inline webapp/src/containers/createRequest/index.tsx:172]:
```suggestion
                                        className='color-link btn btn-primary servicenow-request-button'
```
util classes can follow kebab-case

Comment (ayusht2810) [reply webapp/src/containers/createRequest/index.tsx:172]: i think this is an already defined mattermost class

File: server/plugin/command.go (hunk 3 above)
Comment (ayusht2810) [reply server/plugin/command.go:281]: please check this comment
https://github.com/mattermost/mattermost-plugin-servicenow/pull/78#discussion_r996857980

**Review by avas27JTG — APPROVED** [review]

**Pull request merged**